R/estimateDivergence.R

```diff
@@ -0,0 +1,147 @@
+#'   
+#' @param method a method that is used to calculate the distance. Method is passed to the
+#'   function that is specified by \code{FUN}. By default, \code{method} is \code{"bray"}.
+#'
+#' @param BPPARAM A
+#'   \code{\link[BiocParallel:BiocParallelParam-class]{BiocParallelParam}}
+#'   object specifying whether calculation of estimates should be parallelized.
+#'
+#' @param ... optional arguments
+#' 
+#' @return \code{x} with additional \code{\link{colData}} named \code{*name*}
+#' 
+#' @details
+#'
+#' Microbiota divergence (heterogeneity / spread) within a given sample
+#' set can be quantified by the average sample dissimilarity or beta
+#' diversity with respect to a given reference sample.
+#'
+#' This measure is sensitive to sample size.
+#' Subsampling or bootstrapping can be applied to equalize sample sizes
+#' between comparisons.
+#' 
+#' @seealso
+#' \code{\link[scater:plotColData]{plotColData}}
+#' \itemize{
+#'   \item{\code{\link[mia:estimateRichness]{estimateRichness}}}
+#'   \item{\code{\link[mia:estimateEvenness]{estimateEvenness}}}
+#'   \item{\code{\link[mia:estimateDominance]{estimateDominance}}}
+#'   \item{\code{\link[mia:calculateDistance]{calculateDistance}}}
+#' }
+#' 
+#' @name estimateDivergence
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#' 
+#' @examples
+#' data(GlobalPatterns)
+#' tse <- GlobalPatterns
+#' 
+#' # By default, reference is median of all samples. The name of column where results
+#' # is "divergence" by default, but it can be specified. 
+#' tse <- estimateDivergence(tse)
+#' 
+#' # The method that are used to calculate distance in divergence and 
+#' # reference can be specified. Here, euclidean distance and dist function from 
+#' # stats package are used. Reference is the first sample.
+#' tse <- estimateDivergence(tse, name = "divergence_first_sample", 
+#'                           reference = assays(tse)$counts[,1], 
+#'                           FUN = stats::dist, method = "euclidean")
+#' 
+#' # Reference can also be median or mean of all samples. 
+#' # By default, divergence is calculated by using median. Here, mean is used.
+#' tse <- estimateDivergence(tse, name = "divergence_average", reference = "mean")
+#' 
+#' # All three divergence results are stored in colData.
+#' colData(tse)
+#' 
+NULL
+
+#' @rdname estimateDivergence
+#' @export
+setGeneric("estimateDivergence",signature = c("x"),
+           function(x, abund_values = "counts", name = "divergence", 
+                    reference = "median", FUN = vegan::vegdist, method = "bray", 
+                    ...)
+             standardGeneric("estimateDivergence"))
+
+#' @rdname estimateDivergence
+#' @export
+setMethod("estimateDivergence", signature = c(x="SummarizedExperiment"),
+    function(x, abund_values = "counts", name = "divergence", 
+             reference = "median", FUN = vegan::vegdist, method = "bray", 
+             ..., BPPARAM = SerialParam()){
```

Comment:
`BPPARAM` is not used. Please remove the argument or use it. If you remove it, please also remove the documentation.

Reply:
I used `estimateDiversity` as a template, so it was left there by mistake

Removed